Add admission logic to v_replica_set_controller

fredred375 opened the pull request:
@marshtompsxd  Implemented admission specs stated [here](https://github.com/anvil-verifier/anvil/blob/main/src/v2/controllers/vreplicaset_controller/trusted/spec_types.rs#L115-L128) in the form of a admission webhook server.

marshtompsxd commented:
@fredred375 Thank you! This is a great start! Could you do the following:
- Add a README in this folder on how to build, deploy, and use this admission control
- Add a test case to validate this admission control's implementation (see examples here https://github.com/anvil-verifier/anvil/blob/main/e2e/src/vreplicaset_e2e.rs)

fredred375 commented:
@marshtompsxd 
- Added readme, manifests, and scripts for instructions on local deployment
- Added test to run through all yaml files in `manifests/replicaset/`
  - Only manifests starting with "ok" is expected to pass

fredred375 commented:
@marshtompsxd

Rebased

marshtompsxd commented:
@fredred375 Thanks! One last thing: can you add the test of the admission control of vreplicaset to our CI: https://github.com/anvil-verifier/anvil/blob/main/.github/workflows/ci.yml? See this example https://github.com/anvil-verifier/anvil/blob/main/.github/workflows/ci.yml#L368-L386

fredred375 commented:
@marshtompsxd

Added admission tests into ci
